javatests/arcs/core/host/TestConstructedParticle.kt

```diff
@@ -0,0 +1,23 @@
+package arcs.core.host
+
+import arcs.core.data.Plan
+import arcs.core.data.Schema
+import arcs.core.entity.EntityBaseSpec
+import arcs.sdk.BaseParticle
+import arcs.sdk.HandleHolderBase
+
+fun build(spec: Plan.Particle): TestConstructedParticle {
```

Comment:
Can we now move from this build function to the particle accepting Plan.Particle in the constructor? Then we can pass the class constructor (as opposed to build function) in particle registration, which IMO is nicer, WDYT?